### DISCUSSION TITLE
Create options to change the icons of apps in taskbar

### DISCUSSION THREAD

**Discussion by ritulahkar** [linuxmint/discussions#321] [Ideas]
There are various ways of installing apps, Making is confusing to know which app is flatpak, which is system problematic. And some apps don't have own icon in the taskbar like many appimages. So, ability to show different icons depending on the executable running will be very helpful.

**Comment by anaximeno** [comment]
Can  you provide an example (e.g. screenshot or something like that)?

**Reply by ritulahkar** [reply]
Thank you for your reply.

For example, I downloaded the [PDF4QT Version 1.3.7](https://github.com/JakubMelka/PDF4QT/releases/tag/v1.3.7) AppImage. I made it executable and added to the menu with menu item editor like the following.
![image](https://github.com/linuxmint/discussions/assets/105012118/e2e831ed-0c01-445f-8d9b-d17a486e9e94)
You can see the PDF4QT in the list.

When I launch the app it does not set its icon so a generic icon of wheel in a box comes in taskbar, as follows.
![image](https://github.com/linuxmint/discussions/assets/105012118/643b63b7-9a57-499c-9a52-153af7759cd8)

See the icon in the image.


Now what I want is a way to tell cinnamon that if you are running this executable show this icon in the taskbar. There should be an option to select whether to show the icon the app sets or some custom icon. This should be editable like the menu editing.

**Comment by PWungsten** [comment]
I have the same problem with a program I am developing. Mint 22.2 Cinnamon.

**Comment by klangman** [comment]
I believe this is a problem in the .desktop file for PDF4QT.  Maybe it's missing the `StartupWMClass=` entry?

1. With PDF4QT running type `xprop WM_CLASS` in a terminal and then click on the PDF4QT window.
2. Edit the PDF4QT .desktop file and add a `StartupWMClass=` line with the `xprop WM_CLASS` output after the `=`

After that, the window list should be able to associate the running PDF4QT program with the .desktop file and show the correct icon.

Hope this works for you.
